### ISSUE TITLE
module fail with worker tier

### ISSUE THREAD

**Issue by AlmogCohen** [cloudposse/terraform-aws-elastic-beanstalk-environment#83]
Running the example code with the additional `tier      = "Worker"` option fail with:

```

Error: Error applying plan:

1 error(s) occurred:

* module.elastic_beanstalk_environment.aws_elastic_beanstalk_environment.default: 1 error(s) occurred:

* aws_elastic_beanstalk_environment.default: ConfigurationValidationException: Configuration validation exception: Load Balancer ListenerEnabled setting cannot be applied because AWSEBLoadBalancer doesn't exist.
	status code: 400, request id: bd7a5475-7390-4fe9-a213-a728bf9c3895
```

Manually commenting out all the eb settings that starts with `aws:elb` worked. If I can come up with some better solution I'll propose a PR.

**Comment by AlmogCohen** [comment]
While testing I see that even setting the `ListenerEnabled` settings manually to `"false"` yields the same exception. That means that the only way is currently (without using Terraform 0.12 [dynamic blocks](https://www.terraform.io/docs/configuration/expressions.html#dynamic-blocks) is to remove support for worker tier for this module, as it doesn't work along with the load balancer settings.

**Comment by AlmogCohen** [comment]
For now, I simply created a fork that is suitable for worker environments. It is tested to work when I change the module pointer to the `worker-environment` branch in my fork:
```
module "elastic_beanstalk_environment" {
  source    = "git::https://github.com/AlmogCohen/terraform-aws-elastic-beanstalk-environment.git?ref=worker-environment"
...
}
```

Hope to find a better way to fix this...

**Comment by AlmogCohen** [comment]
Maybe a sister module for workers is needed instead... :)

**Comment by Rodeoclash** [comment]
eI just ran into this problem too, however it seemed that removing anything to do with the elb (even commands that disable it) seems to allow this to work. Perhaps try that?

**Comment by ngealy** [comment]
I opened PR #93 to fix this using Terraform 0.12 and dynamic blocks
